inference-engine/tests/functional/plugin/cpu/single_layer_tests/concat.cpp

```diff
@@ -89,151 +103,510 @@ const auto blocked16_4D_ref = CPUSpecificParams{{nChw16c}, {nChw16c}, {}, "ref"}
 
-INSTANTIATE_TEST_SUITE_P(smoke_Concat5D_CPU_Block16, ConcatLayerCPUTest,
+INSTANTIATE_TEST_SUITE_P(smoke_Concat5D_CPU_Block16_static, ConcatLayerCPUTest,
                         ::testing::Combine(
                                 ::testing::Values(2, 3, 4),
-                                ::testing::Values(std::vector<std::vector<size_t>>{{2, 32, 3, 5, 7},
-                                                                                   {2, 32, 3, 5, 7}}),
+                                ::testing::Values(static_shapes_to_test_representation({{2, 32, 3, 5, 7}, {2, 32, 3, 5, 7}})),
                                 ::testing::ValuesIn(netPrecisions),
-                                ::testing::Values(CommonTestUtils::DEVICE_CPU),
                                 ::testing::Values(blocked16_5D_ref)),
                         ConcatLayerCPUTest::getTestCaseName);
 
+const std::vector<std::vector<InputShape>> inputShapes5D_Block_axis1 = {
+        {
+            {{-1, 32, -1, -1, -1}, {{2, 32, 5, 7, 6},  {1, 32, 10, 2, 8}, {3, 32, 1, 8, 10}}},
+            {{-1, 16, -1, -1, -1}, {{2, 16, 5, 7, 6},  {1, 16, 10, 2, 8}, {3, 16, 1, 8, 10}}},
+            {{-1, 64, -1, -1, -1}, {{2, 64, 5, 7, 6}, {1, 64, 10, 2, 8}, {3, 64, 1, 8, 10}}},
+        },
+        {
+            {{{1, 3}, 32, {1, 10}, {2, 8}, {6, 10}}, {{2, 32, 5, 7, 6}, {1, 32, 10, 2, 8}, {3, 32, 1, 8, 10}}},
+            {{{1, 3}, 16, {1, 10}, {2, 8}, {6, 10}}, {{2, 16, 5, 7, 6}, {1, 16, 10, 2, 8}, {3, 16, 1, 8, 10}}},
+            {{{1, 3}, 64, {1, 10}, {2, 8}, {6, 10}}, {{2, 64, 5, 7, 6}, {1, 64, 10, 2, 8}, {3, 64, 1, 8, 10}}},
+        },
+};
 
-INSTANTIATE_TEST_SUITE_P(smoke_Concat_inPlace, ConcatLayerCPUTest,
+INSTANTIATE_TEST_SUITE_P(smoke_Concat5D_CPU_Block_dynamic_axis_1, ConcatLayerCPUTest,
                         ::testing::Combine(
-                                ::testing::Values(0, 1, 2),
-                                ::testing::Values(std::vector<std::vector<size_t>>{{1, 1, 1, 10},
-                                                                                   {1, 1, 1, 10}},
-                                                  std::vector<std::vector<size_t>>{{1, 1, 5},
-                                                                                   {1, 1, 5}}),
+                                ::testing::Values(1),
+                                ::testing::ValuesIn(inputShapes5D_Block_axis1),
                                 ::testing::ValuesIn(netPrecisions),
-                                ::testing::Values(CommonTestUtils::DEVICE_CPU),
-                                ::testing::Values(CPUSpecificParams{{}, {}, {}, "unknown"})),
+                                ::testing::Values(blocked8_5D_ref, blocked16_5D_ref)),
+                        ConcatLayerCPUTest::getTestCaseName);
+
+const std::vector<std::vector<InputShape>> inputShapes5D_axis1 = {
+        {
+            {{-1, -1, -1, -1, -1}, {{2, 5, 5, 7, 6}, {1, 3, 10, 2, 8}, {3, 4, 1, 8, 10}}},
```

Comment:
Do we cover cases then Concat has two/three inputs with different dynamic shapes (e.g. (-1, -1, -1, -1), (-1, 10, -1, -1)?Oven the case there one of  inputs is dynamic and the second one is static?

Reply:
I guess it's general question for all nodes.
According to spec concat must have equal shape on all position exclude axis position. So in fact if one of the inputs is static the other should be static in all position exclude probably axis.

I will add test cases when axis shape dynamic and static on different inputs. And case when one inputs fully static and other dynamic.